### PR TITLE
Private-events project Showcasing Associations

### PR DESCRIPTION
**Pull request by Emmanuelaaron**
In this project, 
 - we have implemented associations in ruby on rails
 - Written both unit tests and integration tests using RSpec and capybara respectively

### CODE REVIEW

**Comment by Anna-Myzukina** [comment]
### You did a great job so far, but there are a few more things required.


## [REQUIRED]


#### Models :

##### Event model should include `upcoming` and `previous` events


* [x] scope for upcoming events **(covered by automated tests)** from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations)
screenshot 
```ruby
scope :upcoming, -> { where('date >= ?', Time.zone.now) }
```

* [ ] scope for previous events **(covered by automated tests)**  from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations)
screenshot
```ruby
scope :previous, -> { where('date < ?',  Time.zone.now) }
```


#### Associations :

##### User as event creator

* [x] User model: `has_many :events, foreign_key: :creator_id`,  from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations)


screenshot
* [x] Event model `belongs_to :creator, class_name: "User"`,  from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations)


screenshot
  * Migrations
    * [x] events table has creator_id as the foreign key,  from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations)


screenshot

```ruby
def change
  create_table :users do |t|
  end

  create_table :events do |t|
    t.integer :creator_id
  end
end
```

##### User as an Event attendee,  from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations)


screenshot
* [x] There are 2 equally good options to achieve that, described here https://guides.rubyonrails.org/association_basics.html#choosing-between-has-many-through-and-has-and-belongs-to-many

#### Controllers

##### EventsController `create` action :

* [x] uses #build method on association, i.e. `current_user.events.build(event_params)` , from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations)


screenshot 
##### UsersController `show` action:

* [x] lists events created by user, i.e. `@created_events = current_user.events` , from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations)


screenshot 
* [x] lists events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous` , from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations)


screenshot 
* [x] lists upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ,  from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations)


screenshot

### ! Please fix linter, it`s not working yet.
### Make changes and submit another code review request.

*If you have any questions, let me know on slack (@Anna-Muzykina) and I will be glad to clarify with you.*
*Happy coding * <g-emoji class="g-emoji" alias="computer" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f4bb.png">computer</g-emoji>

File: db/migrate/20200717055655_create_attendances.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class CreateAttendances < ActiveRecord::Migration[6.0]
+  def change
+    create_table :attendances do |t|
+      t.integer :attended_event_id
+      t.integer :event_attendee_id
+
+      t.timestamps
```
Comment (maelfosso) [inline db/migrate/20200717055655_create_attendances.rb:7]: add indexes here!

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
+class EventsController < ApplicationController
+  before_action :authenticate_user!, only: %i[new create index show]
+  def new
+    @event = Event.new
+  end
+
+  def invite
+    # debugger
```
Comment (maelfosso) [inline app/controllers/events_controller.rb]: Remove the comment

**Review by maelfosso — APPROVED** [review]
# Congratulations :clap: 

Thanks for applying the suggested changes
You can merge it now :+1: 

**PS** Before merging read, apply the comments left below

Good continuation :handshake: 

Mael FOSSO

**Pull request merged**